pkg/services/sqlstore/preferences.go

```diff
@@ -114,6 +115,7 @@ func (ss *SQLStore) SavePreferences(ctx context.Context, cmd *models.SavePrefere
 		prefs.Timezone = cmd.Timezone
 		prefs.WeekStart = cmd.WeekStart
 		prefs.Theme = cmd.Theme
+		prefs.QueryHistory = cmd.QueryHistory
```

Comment:
Something to keep in mind that if `QueryHistory` is not set (which it's not for any existing API integration) it will clear the `QueryHistory` column. That's because `AllCols()` being used below. 

So to be backward compatible you might need to remove the `AllCols` and specify all the columns you want to be updated using `MustCols(...)` except `QueryHistory`- if `QueryHistory` is nil/empty it will not be updated in the database. More info https://gobook.io/read/gitea.com/xorm/manual-en-US/chapter-06/index.html